mm2src/lp_swap/maker_swap.rs

```diff
@@ -717,7 +798,8 @@ impl MakerSwap {
             .validate_taker_payment(
                 &self.r().taker_payment.clone().unwrap().tx_hex,
                 self.taker_payment_lock.load(Ordering::Relaxed) as u32,
-                &*self.r().other_persistent_pub,
+                &*self.r().other_taker_coin_htlc_pub,
+                &**self.r().taker_coin_htlc_keypair.public(),
```

Comment:
I think it's a good time to add a struct that consists of the method arguments.
It's easy to make a mistake, for example, pass maker and taker pubkeys in the wrong order.

Reply:
I agree, will do it tomorrow.